whatis277/harvest
# bean: Add production related code

Adds production Dockerfile

Renames tern config to non-hidden files
They were excluded by the Dockerfile for builds
Allows us to copy the migration config file to the prod image

Adds TLS support for Postgres and Redis

Testing instructions:
Integration tests:
1. Run bean

    ```bash
    > dc up --build bean
    ```

1. Run the tests

    ```bash
    dc exec -e INTEGRATION=1 bean go test ./...
    ```

Testing TLS:
1. Setup databases on DigitalOcean:
    * Postgres
    * Redis

1. Update the environment variables to use them

1. Run bean
    
    ```bash
    > dc up --build bean
    ```

1. Ensure logs show the server is listening

Testing migration conf file:
1. Run bean

    ```bash
    > dc up --build bean
    ```
    
1. Ensure the file exists

    ```bash
    > dc exec bean cat /go/src/bean/config/migration.tern.conf
    ```
